documentation/src/main/asciidoc/cassandraclient.adoc

```diff
@@ -0,0 +1,299 @@
+= Quarkus - Using the Cassandra Client
+
+Cassandra is a well known NoSQL Database that is widely used.
+
+In this guide, we will see how you can get your REST services to use a Cassandra database.
+
+== Prerequisites
+
+To complete this guide, you need:
+
+* an IDE
+* JDK 1.8+ installed with `JAVA_HOME` configured appropriately
+* GraalVM installed with `GRAALVM_HOME` configured appropriately if you want to use the native mode.
+* Apache Maven 3.5.3+
+* Cassandra installed or Docker installed
+
+== Architecture
+
+The application built in this guide is quite simple: the user can add elements in a list using a form and the list is updated.
+
+All the information between the browser and the server is formatted as JSON.
+
+The elements are stored in the Cassandra Database.
+
+
+== Solution
+
+We recommend that you follow the instructions in the next sections and create the application step by step.
+However, you can go right to the completed example.
+
+The solution is located in the `quickstart` {quickstart-clone-url}[directory].
+
+== Creating the Maven project
+
+First, we need a new project. Create a new maven project and copy the `pom.xml` file that is present in the `quickstart` directory.
+
+The `pom.xml` is importing the RESTEasy/JAX-RS, JSON-B, Context Propagation and Cassandra Client extensions.
+We will be building a REST application using the DataStax Object Mapper {mapper-docs-url}[DataStax Object Mapper].
+It will simplify the Data Access Layer code.
+
+The most important part of the `pom.xml` is adding the `cassandra-quarkus` extension:
+[source,xml]
+----
+<dependency>
+  <groupId>com.datastax.oss.quarkus</groupId>
+  <artifactId>cassandra-quarkus-client</artifactId>
+  <version>${quarkus.version}</version>
+</dependency>
+----
```

Comment:
Reminder to maybe update this if we decide to switch to the BOM (per other private discussion).

Reply:
Getting back to this discussion, I am hesitating about adding it. 
By using our bom, we will get rid of only one dependency, that is:
```
  <dependency>
      <groupId>com.datastax.oss.quarkus</groupId>
      <artifactId>cassandra-quarkus-client</artifactId>
      <version>${project.version}</version>
    </dependency>
```
by replacing it with:
```
<dependencyManagement>
    <dependencies>
      <dependency>
        <groupId>io.quarkus</groupId>
        <artifactId>cassandra-quarkus-bom</artifactId>
        <version>${project.version}</version>
        <type>pom</type>
        <scope>import</scope>
      </dependency>
    </dependencies>
  </dependencyManagement>
```
I am not sure if that is a good solution. I imagine that not all users are familiar with boms. Also, other quick start guides are using direct dependency to extension, instead of the bom (they are using bom internally in the extension as well)

Reply:
Frankly I think these boms are a bit overkill. They contain 3/4 artifacts each. That's not enough to justify their presence.

Reply:
Leaving as is for now.